spring-integration-core/src/main/java/org/springframework/integration/gateway/MessagingGatewaySupport.java

```diff
@@ -690,6 +726,25 @@ private void sendMessageForReactiveFlow(MessageChannel requestChannel, Message<?
 		}
 	}
 
+	protected TimerFacade sendTimer() {
+		if (this.successTimer == null) {
+			this.successTimer = buildSendTimer(true, "none");
+		}
+		return this.successTimer;
+	}
+
+	protected TimerFacade buildSendTimer(boolean success, String exception) {
+		TimerFacade timer = this.metricsCaptor.timerBuilder(SEND_TIMER_NAME)
+				.tag("type", "source")
```

Comment:
Do we really talk here about `source`?

Reply:
It really is a message source - I thought it was better than introducing a new type `gateway` because that can be in or out.